vars/capsulePipelineWrapper.groovy

```diff
@@ -1,245 +1,485 @@
+      branch: 'main',
+      directory: version,
+      credentialsId: ghCredentialsID
+  ])
+  sh label: 'Building ' + binType,
+        script: 'go mod vendor && go build -o ' + outDir + '/' + binType + ' ' + packagesMapping[binType]
+}
+
+void downloadS3Binary(String s3Path, String binName, String outDir) {
+  sh label: 'Download the ' + binName + ' from the S3',
+       script: 'aws s3 cp --only-show-errors --no-progress "' + s3Path + '" "' + outDir + '/' + binName + '"'
+  sh label: 'Set permissions on the ' + binName + ' binary',
+       script: 'chmod a+x "' + outDir + '/' + binName + '"'
+}
+
+boolean isS3Link(String path) {
+    /* groovylint-disable-next-line DuplicateNumberLiteral */
+  return path.length() > 0 && path[0..4] == 's3://'
+}
+
+void veganetSh(String network, String command, String sshCredentialsId, String dockerCredentialsId) {
+  dir('devops-infra') {
+    withCredentials([sshUserPrivateKey(
+        credentialsId: sshCredentialsId,
+        keyFileVariable: 'PSSH_KEYFILE',
+        usernameVariable: 'PSSH_USER'
+    )]) {
+      withDockerRegistry([
+          credentialsId: dockerCredentialsId,
+          url: 'https://ghcr.io'
+      ]) {
+        withGoogleSA('gcp-k8s') {
+          sh label: 'Create markets',
+            script: './veganet.sh ' + network + ' ' + command
+        }
+      }
     }
+  }
+}
+
+void call(Map customConfig = [:]) {
+  Map config = [
+    networkName: '',
+    nomadAddress: '',
+    awsRegion: '',
+    vegacapsuleS3BucketName: '',
+  ] + customConfig
+
+  pipeline {
+    agent any
+    environment {
+      // TO-DO: Add secrets to jenkins
+      NOMAD_TLS_SERVER_NAME = 'server.global.nomad'
+      NOMAD_CACERT = credentials('nomad-cacert')
+      NOMAD_CLIENT_CERT = credentials('nomad-client-cert')
+      NOMAD_CLIENT_KEY = credentials('nomad-client-key')
+      NOMAD_ADDR = "${config.nomadAddress}"
+
+      AWS_SECRET_ACCESS_KEY = credentials('jenkins-vegacapsule-aws-secret')
+      AWS_ACCESS_KEY_ID = credentials('jenkins-vegacapsule-aws-id')
+      S3_BUCKET_NAME = "${config.vegacapsuleS3BucketName}"
+      AWS_REGION = "${config.awsRegion}"
 
-    pipeline {
-        agent any
-        environment {
-            // TO-DO: Add secrets to jenkins
-            NOMAD_TLS_SERVER_NAME = "server.global.nomad"
-            NOMAD_CACERT = credentials('nomad-cacert')
-            NOMAD_CLIENT_CERT = credentials('nomad-client-cert')
-            NOMAD_CLIENT_KEY = credentials('nomad-client-key')
+      PATH = "${env.WORKSPACE}/bin:${env.PATH}"
+      GITHUB_CREDS = 'github-vega-ci-bot-artifacts'
+      GITHUB_SSH_CREDS = 'vega-ci-bot'
+      NETWORK_SSH_CREDENTIALS = 'ubuntu-ansible-key'
+    }
 
-            AWS_SECRET_ACCESS_KEY = credentials('jenkins-vegacapsule-aws-secret')
-            AWS_ACCESS_KEY_ID = credentials('jenkins-vegacapsule-aws-id')
+    stages {
+      stage('Prepare') {
+        steps {
+          script {
+            print('Parameters: ' + params)
```

Comment:
Nit: if you do echo "parameters: ${params}" you don't need script block, as echo is step where print is groovy function. in most cases Jenkins prefers steps over functions